.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,25 @@
+Thank you for submitting the pull request to the Apache Ignite.
+
+In order to streamline the review of the contribution 
+we ask you to ensure the following steps have been taken:
+
+### The Contribution Checklist
+- [ ] The JIRA ticket has been associated with the pull request
+- [ ] The JIRA ticket has the _Patch Available_ state
+- [ ] The pull request briefly describes changes that have been made
+- [ ] The pull request title has the following pattern 
+e.g. `IGNITE-12407: Java thin client. Cluster API support`
+- [ ] Commits have the following pattern 
```

Comment:
In my opinion, this should not be a requirement. We merge with squash, why waste time on intermediate commits? There can be hundreds of them with review fixes, test tweaks, and so on.

Reply:
I've removed this point.